_posts/pl/2022-12-01-memory-java.md

```diff
@@ -0,0 +1,241 @@
+
+-                      Code (reserved=255775KB, committed=103615KB)
+                            (malloc=8087KB #29935)
+                            (mmap: reserved=247688KB, committed=95528KB)
+
+-                        GC (reserved=2104KB, committed=1908KB)
+                            (malloc=1032KB #4341)
+                            (mmap: reserved=1072KB, committed=876KB)
+
+-                  Compiler (reserved=1046KB, committed=1046KB)
+                            (malloc=914KB #2701)
+                            (arena=133KB #5)
+
+-                  Internal (reserved=1198KB, committed=1198KB)
+                            (malloc=1166KB #2425)
+                            (mmap: reserved=32KB, committed=32KB)
+
+-                     Other (reserved=302KB, committed=302KB)
+                            (malloc=302KB #39)
+
+-                    Symbol (reserved=28055KB, committed=28055KB)
+                            (malloc=25969KB #368221)
+                            (arena=2086KB #1)
+
+-    Native Memory Tracking (reserved=8867KB, committed=8867KB)
+     (malloc=509KB #7225)
+     (tracking overhead=8358KB)
+
+-        Shared class space (reserved=17148KB, committed=17148KB)
+                            (mmap: reserved=17148KB, committed=17148KB)
+
+-               Arena Chunk (reserved=179KB, committed=179KB)
+                            (malloc=179KB)
+
+-                   Logging (reserved=4KB, committed=4KB)
+                            (malloc=4KB #192)
+
+-                 Arguments (reserved=19KB, committed=19KB)
+                            (malloc=19KB #531)
+
+-                    Module (reserved=1365KB, committed=1365KB)
+                            (malloc=1365KB #6576)
+
+-              Synchronizer (reserved=594KB, committed=594KB)
+                            (malloc=594KB #5027)
+
+-                 Safepoint (reserved=8KB, committed=8KB)
+                            (mmap: reserved=8KB, committed=8KB)
+```
+Najbardziej interesujące są tutaj wartości `commited`, pokazujące faktycznie używaną pamięć.
+
+Co możemy z tego zrzutu wyczytać? Przede wszystkim, że z punktu widzenia samego JVMa, proces zajmuje 610501KB, czyli około 596MB pamięci.
+
+Ciekawymi wartościami są też te dotyczące heapa oraz wszystkich pozostałych rzeczy składających się na off-heap. Warto zdawać sobie sprawę, że offheap bez problemu potrafi wciągnąć kilkaset MB i trzeba to wziąć pod uwagę w trakcie strojenia pamięci. Nie będziemy się tu jednak wgłębiać we wszystkie składowe off heapa.
+
+Spójrzmy natomiast na wyjście ps aux dotyczące tej samej aplikacji.
+
+```
+USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
+1000630+       1  0.8  2.0 2283920 721032 ?      Ssl  Nov22  12:27 Java -Dspring.config.location=classpath:/applic....
+I tutaj widzimy drugi rozjazd - według NMT proces bierze niecałe 600MB RAMu, podczas gdy według systemu operacyjnego jest to około 700MB. Gdzie podziało się 100MB?
```

Comment:
to zdanie poza blok kodu musi wylecieć ;)

Reply:
done